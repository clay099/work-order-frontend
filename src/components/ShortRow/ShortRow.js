import React from "react";
import { TableCell } from "@material-ui/core/";
import dateHelper from "../../helperFunctions/dateHelper";

const ShortRow = ({ project }) => {
	return (
		<>
			{project.description ? (
				<TableCell>{project.description}</TableCell>
			) : (
				<TableCell>not available</TableCell>
			)}
			{project.street_address ? (
				<TableCell>{project.street_address}</TableCell>
			) : (
				<TableCell>not available</TableCell>
			)}
			{project.address_city ? (
				<TableCell>{project.address_city}</TableCell>
			) : (
				<TableCell>not available</TableCell>
			)}
			{project.created_at ? (
				<TableCell>{dateHelper(project.created_at)}</TableCell>
			) : (
				<TableCell>not available</TableCell>
			)}
			{project.status ? (
				<TableCell>{project.status}</TableCell>
			) : (
				<TableCell>no data</TableCell>
			)}
		</>
	);
};

export default ShortRow;
