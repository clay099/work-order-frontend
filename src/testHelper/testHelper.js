// testHelper.js
import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { createStore } from "redux";
import { Provider } from "react-redux";
// Import your own reducer
import reducer from "../reducers/rootReducer";

function render(
	ui,
	{
		initialState,
		store = createStore(reducer, initialState),
		...renderOptions
	} = {}
) {
	function Wrapper({ children }) {
		return <Provider store={store}>{children}</Provider>;
	}
	return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const tokenState = { token: "token" };
const userTypeUserState = { userType: "user" };
const userTypeTradesmenState = { userType: "user" };
const idState = { id: 1234 };
const userDetailsState = { userDetails: "userDetails" };
const handleSubmitState = { handleSubmit: jest.fn() };
const handleChangeState = { handleChange: jest.fn() };
const disabledState = { disabled: false };
const disabledEmailState = { disabledEmail: false };
const buttonTextState = { buttonText: "buttonText" };
const formDataState = {
	formData: {
		id: 1,
		firstName: "firstName",
		lastName: "lastName",
		email: "email",
		password: "password",
		newPassword: "newPassword",
		confirmNewPassword: "confirmNewPassword",
	},
};

// re-export everything
export * from "@testing-library/react";
// override render method
export {
	render,
	tokenState,
	userTypeUserState,
	userTypeTradesmenState,
	idState,
	userDetailsState,
	handleSubmitState,
	handleChangeState,
	disabledState,
	disabledEmailState,
	buttonTextState,
	formDataState,
};
