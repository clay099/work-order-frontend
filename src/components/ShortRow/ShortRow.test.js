import React from "react";
import ShortRow from "./ShortRow";
import { render } from "../../testHelper/testHelper";
console.error = jest.fn();

const params = {
	project: {
		description: "description",
		street_address: "street_address",
		address_city: "address_city",
		created_at: "created_at",
		status: "status",
	},
};

describe("ShortRow Component", () => {
	// smoke test
	test("renders without crashing", () => {
		render(<ShortRow {...params} />);
	});

	// snapshot test
	test("matches snapshot", () => {
		const { asFragment } = render(<ShortRow {...params} />);
		expect(asFragment).toMatchSnapshot();
	});

	test("value to be in the document", () => {
		const { getByText } = render(<ShortRow {...params} />);
		const value = getByText(params.project.description);
		expect(value).toBeInTheDocument();
	});
});
