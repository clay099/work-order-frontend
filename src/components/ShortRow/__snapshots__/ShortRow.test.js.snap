// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShortRow Component matches snapshot 1`] = `[Function]`;
